app/Http/Forms/PersonalDataForm.php

```
@@ -0,0 +1,168 @@
+				'error_messages' => [
+					'required_with' => trans('payment.invoice-required')
+				],
+				'label' => trans('payment.invoice-address'),
+				'rules' => 'required_with:invoice',
+				'attr'  => [
+					'class' => 'input',
+					'placeholder' => trans('payment.invoice-address'),
+				],
+			])
+			->add('invoice_zip', 'text', [
+				'error_messages' => [
+					'required_with' => trans('payment.invoice-required')
+				],
+				'label' => trans('payment.invoice-zip'),
+				'rules' => 'required_with:invoice',
+				'attr'  => [
+					'class' => 'input',
+					'placeholder' => trans('payment.invoice-zip'),
+				],
+			])
+			->add('invoice_city', 'text', [
+				'error_messages' => [
+					'required_with' => trans('payment.invoice-required')
+				],
+				'label' => trans('payment.invoice-city'),
+				'rules' => 'required_with:invoice',
+				'attr'  => [
+					'class' => 'input',
+					'placeholder' => trans('payment.invoice-city'),
+				],
+			])
+			->add('invoice_country', 'text', [
+				'error_messages' => [
+					'required_with' => trans('payment.invoice-required')
+				],
+				'label' => trans('payment.invoice-country'),
+				'rules' => 'required_with:invoice',
+				'attr'  => [
+					'class' => 'input',
+					'placeholder' => trans('payment.invoice-country'),
+				],
+			]);
+	}
+
+	public function validate($validationRules = [], $messages = [])
+	{
+		$validator = $this->getIdentityNumberValidator($this->getRequest()->get('identity_number_type'));
+
+		if ($validator && !$this->getField('identity_number')->getOption('attr.disabled')) {
+			$validationRules['identity_number'] = $validator;
+		}
+
+		return parent::validate($validationRules, $messages);
+	}
+
+	protected function getIdentityNumberValidator($identityNumberType) {
+		$validators = [
+			'passport_number' => new ValidatePassportNumber,
+			'personal_identity_number' => new ValidatePersonalIdentityNumber,
+		];
+
+		if (!array_key_exists($identityNumberType, $validators)) {
```

We have rule `required|in:personal_identity_number,passport_number` so this should never happen. But if it does, I'd prefer to fail loudly than allow users to save incorrect data.